TrafficCapture/captureKafkaOffloader/src/main/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/KafkaCaptureFactory.java

```diff
@@ -78,14 +81,14 @@ public IChannelConnectionCaptureSerializer createOffloader(String connectionId)
 
     private Callback handleProducerRecordSent(CompletableFuture cf, String recordId) {
```

Comment:
Can you please add a comment referencing some of the guidance that you saw (from your description of the CR).  As a developer, this is where I might naturally want to put retry code.  Understanding what exceptions can happen here and which ones (all?) have been retried will be really helpful to future devs.